src/Compilers/VisualBasic/Test/Semantic/IOperation/IOperationTests_IFieldReferenceExpression.vb

```diff
@@ -130,5 +130,64 @@ IFieldReferenceOperation: C.i As System.Int32 (OperationKind.FieldReference, Typ
 
             VerifyOperationTreeAndDiagnosticsForTest(Of MemberAccessExpressionSyntax)(source, expectedOperationTree, expectedDiagnostics)
         End Sub
+
+        <CompilerTrait(CompilerFeature.IOperation)>
+        <Fact()>
+        Public Sub IFieldReference_SharedField()
+            Dim source = <![CDATA[
+Option Strict On
+Imports System
+
+Module M1
+    Class C1
+        Shared i1 As Integer
+        Shared Sub S2()
+            Dim i2 = i1'BIND:"i1"
+        End Sub
+    End Class
+End Module]]>.Value
+
+            Dim expectedOperationTree = <![CDATA[
+IFieldReferenceOperation: M1.C1.i1 As System.Int32 (Static) (OperationKind.FieldReference, Type: System.Int32) (Syntax: 'i1')
+  Instance Receiver: 
+    null
+]]>.Value
+
+            Dim expectedDiagnostics = String.Empty
+
+            VerifyOperationTreeAndDiagnosticsForTest(Of IdentifierNameSyntax)(source, expectedOperationTree, expectedDiagnostics)
+        End Sub
+
+        <CompilerTrait(CompilerFeature.IOperation)>
+        <Fact()>
+        Public Sub IFieldReference_SharedFieldWithInstanceReceiver()
```

Comment:
>IFieldReference_SharedFieldWithInstanceReceiver [](start = 19, length = 47)

Please make sure we have tests for accessing a field (static and instance) of off a type

Reply:
Done.

---
In reply to: [149249408](https://github.com/dotnet/roslyn/pull/22972#discussion_r149249408) [](ancestors = 149249408)